src/test/scala/uk/gov/nationalarchives/tdr/api/routes/FileRouteSpec.scala

```diff
@@ -50,6 +50,24 @@ class FileRouteSpec extends TestContainerUtils with Matchers with TestRequest {
       expectedResponse.data.get.addFilesAndMetadata should equal(response.data.get.addFilesAndMetadata)
   }
 
+  "The api" should "add files and metadata entries for files and directories" in withContainers {
+    case container: PostgreSQLContainer =>
+      val consignmentId = UUID.fromString("f1a9269d-157b-402c-98d8-1633393634c5")
+      val utils = TestUtils(container.database)
+      (clientSideProperties ++ staticMetadataProperties.map(_.name)).foreach(utils.addFileProperty)
+      utils.createConsignment(consignmentId, userId)
+
+      val response = runTestMutationFileMetadata("mutation_alldata", validUserToken())
+      val distinctDirectoryCount = 3
+      val fileCount = 2
+      val expectedCount = (staticMetadataProperties.size * distinctDirectoryCount) + //Directories have static metadata
```

Comment:
Are these comments necessary?

Reply:
They helped me to count :smile: They're not needed now, I've taken them out.